src/revaultd/mod.rs

```diff
@@ -161,6 +163,28 @@ impl RevaultD {
             self.call("unvaulttx", Some(vec![outpoint, &unvault_tx]))?;
         Ok(())
     }
+
+    pub fn get_spend_tx(
+        &self,
+        inputs: &[String],
+        outputs: &HashMap<String, u64>,
+        feerate: &u8,
+    ) -> Result<SpendTransaction, RevaultDError> {
+        self.call(
+            "getspendtx",
+            Some(vec![json!(inputs), json!(outputs), json!(feerate)]),
+        )
+        .map(|mut res: SpendTransaction| {
+            res.feerate = *feerate;
+            res
+        })
+    }
+
+    pub fn update_spend_tx(&self, psbt: &Psbt) -> Result<(), RevaultDError> {
+        let spend_tx = base64::encode(&consensus::serialize(psbt));
```

Comment:
Small nit: you should use our types from revault_t for the PSBT you know are always going to be of a specific types. It's a nice way to sanity check the tx and to have some more state assumptions that hold for it, should you ever need it